app/services/email/subscription.rb

```diff
@@ -0,0 +1,23 @@
+class Email::Subscription
```

Comment:
This class is mostly overkill. You can redefine the `User#subscribe_to_newsletter` method to

```ruby
def subscribe_to_newsletter
  EmailSubscriptionWorker.perform_async(id)
end
```

you don't need to pass the `MAILCHIMP_NEWSLETTER_LIST_ID` environment variable as the env variable should be available within the sidekiq worker.

Same thing can be said when unsubscribing an email address. The only difference is you'll want to pass in the email to unsubscribe

```ruby
# Passing in the email because it could be an old email we want to unsubscribe
def unsubscribe_to_newsletter(email_to_unsubscribe)
  EmailRemoveSubscriptionWorker.perform_async(id, email_to_unsubscribe)
end
```